ccic/data/dardar.py

```diff
@@ -0,0 +1,293 @@
+
+        # Remap cloud classes
+        labels = remap_cloud_classes(labels, source_height, surface_altitude_source, ALTITUDE_LEVELS)
+
+        # Pick the labels for the corresponding profiles
+        output_shape = resampler.target_area.shape
+        cloud_mask_r = -1 *  np.ones(output_shape, dtype=np.int8)
+        cloud_mask_r.ravel()[target_indices] = cloud_mask[source_indices]
+        labels_r = -1 * np.ones(output_shape + (20,), dtype=np.int8)
+        labels_r.reshape(-1, 20)[target_indices] = labels[source_indices]
+
+
+        target_dataset["altitude"] = (("altitude",), ALTITUDE_LEVELS)
+        target_dataset["altitude"].attrs = {
+            "units": "meters",
+            "positive": "up"
+        }
+        
+        target_dataset["tiwc"] = (("latitude", "longitude", "altitude"), iwc_r)
+        target_dataset["tiwc"].attrs["long_name"] = "Total ice water content"
+        target_dataset["tiwc"].attrs["unit"] = "g m-3"
+
+        target_dataset["tiwp_fpavg"] = (("latitude", "longitude"), iwp_r)
+        target_dataset["tiwp_fpavg"].attrs["long_name"] = "Footprint-averaged total ice water path"
+        target_dataset["tiwp_fpavg"].attrs["unit"] = "g m-3"
+
+        target_dataset["tiwp"] = (("latitude", "longitude"), iwp_r_rand)
+        target_dataset["tiwp"].attrs["long_name"] = "Total ice water path"
+        target_dataset["tiwp"].attrs["unit"] = "g m-3"
+
+        target_dataset["time_dardar"] = (("latitude", "longitude"), time_r)
+
+        target_dataset["cloud_mask"] = (("latitude", "longitude"), cloud_mask_r)
+        target_dataset["cloud_mask"].attrs = {
+            "long_name": "Cloud presence in atmospheric column",
+            "flag_values": "0, 1",
+            "flag_meaning": "no cloud, cloud present",
+            "_FillValue": -1
+        }
+        target_dataset["cloud_class"] = (("latitude", "longitude", "altitude"), labels_r)
+        target_dataset["cloud_class"].attrs = {
+            "long_name": "Resampled DARMASK Categorization Flags",
+            "flag_values": "-2, -1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15",
+            "flag_meanings": (
+                "presence of liquid unknow",
+                "surface and subsurface",
+                "clear sky",
+                "ice clouds",
+                "spherical or 2D ice",
+                "supercooled water",
+                "supercooled and ice",
+                "cold rain",
+                "aerosol",
+                "warm rain",
+                "stratospheric clouds",
+                "highly concentrated ice",
+                "top of convective towers",
+                "liquid cloud",
+                "warm rain and liquid clouds",
+                "cold rain and liquid clouds",
+                "rain may be mixed with liquid",
+                "Multiple scattering due to supercooled water"
+            ),
+            "_FillValue": -1,
+        }
+
+    def get_iwp(self, dataset, above_ground=True):
```

Comment:
I would suggest moving member function, which don't use the self arguments out of the class to make them stand-alone functions on the module level.

It's more of a cosmetic change, but, at least in principle, it makes it easier to reuse the function.

Reply:
Addressed with 771c8bf